spring-kafka/src/test/java/org/springframework/kafka/annotation/KafkaListenerTests.java

```diff
@@ -0,0 +1,234 @@
+		assertThat(this.otherClassLevelListenerWithSinglePublicMethod.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(this.classLevelListenerWithSinglePublicMethodAndPrivateMethod.latch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		assertThat(this.classLevelListenerWithDoublePublicMethod.latch.await(10, TimeUnit.SECONDS)).isFalse();
+		assertThat(this.classLevelListenerWithDoublePublicMethod.latch.getCount()).isEqualTo(ClassLevelListenerWithDoublePublicMethod.INIT_LATCH_COUNT);
+
+		assertThat(this.kafkaListenerEndpointRegistry.getListenerContainer("classAndMethodLevelListener").getGroupId())
+				.isEqualTo("classAndMethodLevelListener");
+		assertThat(this.kafkaListenerEndpointRegistry.getListenerContainer("classLevelListenerWithSinglePublicMethod").getGroupId())
+				.isEqualTo("classLevelListenerWithSinglePublicMethod");
+		assertThat(this.kafkaListenerEndpointRegistry.getListenerContainer("otherClassLevelListenerWithSinglePublicMethod").getGroupId())
+				.isEqualTo("otherClassLevelListenerWithSinglePublicMethod");
+		assertThat(this.kafkaListenerEndpointRegistry.getListenerContainer("classLevelListenerWithDoublePublicMethod").getGroupId())
+				.isEqualTo("classLevelListenerWithDoublePublicMethod");
+		assertThat(this.kafkaListenerEndpointRegistry.getListenerContainer("classLevelListenerWithSinglePublicMethodAndPrivateMethod").getGroupId())
+				.isEqualTo("classLevelListenerWithSinglePublicMethodAndPrivateMethod");
+
+	}
+
+	@Configuration
+	@EnableKafka
+	public static class Config {
+
+		@Autowired
+		EmbeddedKafkaBroker broker;
+
+		@Bean
+		public KafkaListenerContainerFactory<?> kafkaListenerContainerFactory() {
+			ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+			factory.setConsumerFactory(consumerFactory());
+			return factory;
+		}
+
+		@Bean
+		public DefaultKafkaConsumerFactory<Integer, String> consumerFactory() {
+			return new DefaultKafkaConsumerFactory<>(consumerConfigs());
+		}
+
+		@Bean
+		public Map<String, Object> consumerConfigs() {
+			Map<String, Object> consumerProps = KafkaTestUtils.consumerProps(
+					this.broker.getBrokersAsString(),
+					"myDefaultListenerGroup",
+					"false");
+			return consumerProps;
+		}
+
+		@Bean
+		public KafkaTemplate<Integer, String> template() {
+			return new KafkaTemplate<>(producerFactory());
+		}
+
+		@Bean
+		public ProducerFactory<Integer, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(producerConfigs());
+		}
+
+		@Bean
+		public Map<String, Object> producerConfigs() {
+			return KafkaTestUtils.producerProps(this.broker.getBrokersAsString());
+		}
+
+		@Component
+		@KafkaListener(id = "classAndMethodLevelListener", topics = "default-listener.tests")
+		public static class ClassAndMethodLevelListener {
+
+			final CountDownLatch latch = new CountDownLatch(1);
+
+			@KafkaHandler
```

Comment:
Why do we need to test this here?
I think the `@KafkaHandler` is covered in other tests, e.g. `EnableKafkaIntegrationTests`.